### ISSUE TITLE
Support Fish shell

### ISSUE THREAD

**Issue by eastenluis** [tophat/yvm#116]
**Is your feature request related to a problem? Please describe.**
To use `yvm use`, I have to source yvm within my shell environment. However, because yvm shell syntax is not compatible with fish shell,
sourcing will not work properly.

```
> source /usr/local/bin/yvm
/usr/local/bin/yvm (line 25): Unsupported use of '&&'. In fish, please use 'COMMAND; and COMMAND'.
    [ -n "${1-}" ] && [ -x "$(yvm_version_path "$1" 2> /dev/null)"/bin/yarn ]
                    ^
from sourcing file /usr/local/bin/yvm
        called on standard input

source: Error while reading file '/usr/local/bin/yvm'
```

**Describe the solution you'd like**
Provide additional shell script for sourcing yvm in fish shell. 

`virtualenv` uses a similar approach. In its environment directory, has separate activate scripts for bash, fish shell, and windows CMD.

**Describe alternatives you've considered**

**Additional context**

**Comment by eastenluis** [comment]
Starting up a branch to address this issue: https://github.com/eastenluis/yvm/blob/fish-shell-support/src/yvm.fish.

I found fish shell doesn't rely on interactive mode to modify `$PATH`. By default it inherits the `PATH` and you can set it through `set -U PATH ...` or `set -g PATH ...`, which set a custom `$PATH` "universally" (across all fish terminals) or "globally" (only within the current session). But fish shell always (see this issuehttps://github.com/fish-shell/fish-shell/issues/527) creates a "global" `$PATH`, which shadows the universal one I set. 

The recommended approach from shell is to modify `$fish_user_paths` instead, which always prepends to inherited `$PATH`. I will need to update `get-new-path` to accept a custom path in this case.

**Comment by jakebolam** [comment]
@eastenluis want to put up a PR? After #189 goes in (should make it easier after that)

**Comment by eastenluis** [comment]
@jakebolam I had a PR in my repo before but I want to take a different approach now, especially b/c all changes we have made. I'll try when I had time but anyone feel free to pick it up instead.

**Comment by jakebolam** [comment]
no worries, cheers @eastenluis. Want to put a PR up so we can see the changes? (Feel free to comment on the PR things you would like to change)

**Comment by eastenluis** [comment]
Yeah, let me open up a new PR during the vacation. Basic idea will be setting up a custom `yvm use` for fish shell function and calling it from `yvm.sh`. My previous approach is to write a separate `yvm.fish` for the whole `yvm.sh` but it will be hard to maintain all the changes.
